sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace SomeNamespace
+{
+    class Program
+    {
+    }
+
+    static class Helpers1
+    {
+        static void Foo(this Program p) // Noncompliant {{Either move this extension to another namespace or move the method inside the type itself.}}
```

Comment:
The message will not be always correct - if you are extending an interface, you cannot move the method inside (yet). I would rephrase to something like "Move this extension method in another namespace." when extending interface or type that has no declaration (e.g. 3rd party class for example), and "Either move this extension method to another namespace or declare it as an instance method in 'Program'." when extending class.